corehq/apps/export/models/new.py

```diff
@@ -0,0 +1,342 @@
+from itertools import groupby
+from collections import defaultdict, namedtuple
+from couchdbkit import SchemaListProperty, SchemaProperty
+
+from corehq.apps.userreports.expressions.getters import NestedDictGetter
+from corehq.apps.app_manager.dbaccessors import get_built_app_ids_for_app_id
+from corehq.apps.app_manager.models import Application
+from dimagi.utils.couch.database import iter_docs
+from dimagi.ext.couchdbkit import (
+    Document,
+    DocumentSchema,
+    ListProperty,
+    StringProperty,
+    IntegerProperty,
+)
+
+
+Question = namedtuple('Question', [
+    'type',
+    'label',
+    'value',
+    'repeat',
+    'group',
+    'relevant',
+    'required',
+    'tag',
+    'options',
+    'translations',
+    'calculate',
+])
+Question.__new__.__defaults__ = (None,) * len(Question._fields)  # Defaults all fields to None
+
+
+class ExportItem(DocumentSchema):
+    """
+    An item for export.
+    path is a question path like ["my_group", "q1"] or a case property name
+    like ["date_of_birth"].
+    """
+    path = ListProperty()
+    label = StringProperty()
+    last_occurrence = IntegerProperty()
+
+    @classmethod
+    def create_from_question(cls, question, appVersion):
+        return cls(
+            path=_string_path_to_list(question.value),
+            label=question.label,
+            last_occurrence=appVersion,
+        )
+
+    @classmethod
+    def merge(cls, one, two):
+        item = cls(one.to_json())
+        item.last_occurrence = max(one.last_occurrence, two.last_occurrence)
+        return item
+
+
+class ExportColumn(DocumentSchema):
+    item = SchemaProperty(ExportItem)
+    name = StringProperty()
+    path = ListProperty()
+    label = StringProperty()
+    last_occurrence = IntegerProperty()
```

Comment:
`name`, `path`, and `last_occurrence` should all be unnecessary, as those fields are stored on the `ExportItem`.